src/mlpack/methods/cf/cf.hpp

```diff
@@ -258,22 +258,24 @@ class CF
   //! Cleaned data matrix.
   arma::sp_mat cleanedData;
 
-  /**
-   * Helper function to insert a point into the recommendation matrices.
-   *
-   * @param queryIndex Index of point whose recommendations we are inserting
-   *     into.
-   * @param pos Position in list to insert into.
-   * @param neighbor Index of item being inserted as a recommendation.
-   * @param value Value of recommendation.
-   */
-  void InsertNeighbor(const size_t queryIndex,
-                      const size_t pos,
-                      const size_t neighbor,
-                      const double value,
-                      arma::Mat<size_t>& recommendations,
-                      arma::mat& values) const;
-
+  //! Candidate represents a possible recommendation.
+  struct Candidate
+  {
+    //! Value of this recommendation.
+    double value;
+    //! Item of this recommendation.
+    size_t item;
+    //! Trivial constructor.
+    Candidate(double value, size_t item) :
+        value(value),
+        item(item)
+    {};
+    //! Compare the value of two candidates.
+    friend bool operator>(const Candidate& l, const Candidate& r)
+    {
+      return l.value > r.value;
+    };
+  };
```

Comment:
That makes sense---there is not much reason to believe that `std::pair<>` is intrinsically faster than a handwritten class that implements the same functionality.  My leaning towards `pair` is primarily a function of minimizing code, but it is not a huge preference. :)